client/windows/CWindowObject.cpp

```diff
@@ -60,7 +61,8 @@ CWindowObject::CWindowObject(int options_, const ImagePath & imageName):
 	options(options_),
 	background(createBg(imageName, options_ & PLAYER_COLORED))
 {
-	assert(parent == nullptr); //Safe to remove, but windows should not have parent
+	if(!(options & NEEDS_ANIMATED_BACKGROUND)) //currently workaround for highscores (currently uses window as normal control, because otherwise videos are not played in background yet)
+		assert(parent == nullptr); //Safe to remove, but windows should not have parent
```

Comment:
Are you sure that in all release configurations assert never evaluates to nothing causing next statement being under if instead? I am not good at at C++ so I assume that in some cases assert can be a macro.

Reply:
Code is fine. It will evaluate to an empty expression, but this will not change code flow.

Technically in release build condition is redundant, so
`assert(parent == nullptr || (options & NEEDS_ANIMATED_BACKGROUND));`
might be a better option, but both versions will work as expected.